### ISSUE TITLE
Checking request query string?

### ISSUE THREAD

**Issue by michaelherold** [InterNations/http-mock#30]
Is there a way to check the query string of a request (particularly for GET requests)? I want to make sure a transformation in my request wrapper actually occurs when I make the request.

If this isn't yet possible, here's the API I would expect:

```php
$this->http->mock
    ->when()
        ->methodIs("GET")
        ->pathIs("/my/test/path")
        ->queryIs("param1=test1&param2=test2")
    ->then()->body("{}")->statusCode(200)
    ->end();
```

or

```php
$this->http->mock
    ->when()
        ->methodIs("GET")
        ->pathIs("/my/test/path")
        ->queryIs(array("param1" => "test1", "param2" => "test2"))
    ->then()->body("{}")->statusCode(200)
    ->end();
```

PS: Thank you for making this library. It's nice to be able to mock HTTP requests in my test suite.

**Comment by lstrojny** [comment]
Generally you can do anything by passing a matching callback. See this example:

```php
        $this->http->mock
            ->when()
                ->callback(
                    function (Request $request) {
                        return $request->query->has('key1');
                    }
                )
                ->methodIs('GET')
            ->then()
                ->body('query string')
            ->end();
        $this->http->setUp();
```

But you are right, we should have a convenient API to do so. Let me think about it.

**Comment by michaelherold** [comment]
Awesome! Thank you - I missed the callback functionality.

**Comment by lstrojny** [comment]
Was hard to find without reading the code so far. I finally got around adding some documentation. Have a look here, maybe you can find some more gems: https://github.com/InterNations/http-mock/blob/master/doc/index.md

**Comment by lstrojny** [comment]
Have a look at #31! Feedback welcome

**Comment by lstrojny** [comment]
Merged #31
